code/modules/pai/card.dm

```diff
@@ -0,0 +1,264 @@
+			return TRUE
+		if("wipe_pai")
+			pai.wipe_pai(usr)
+			ui.close()
+			return TRUE
+	return FALSE
+
+/** Flashes the pai card screen */
+/obj/item/pai_card/proc/add_alert()
+	if(pai)
+		return
+	add_overlay(
+		list(mutable_appearance(icon, "[initial(icon_state)]-alert"),
+			emissive_appearance(icon, "[initial(icon_state)]-alert", alpha = src.alpha)))
+
+/** Removes any overlays */
+/obj/item/pai_card/proc/remove_alert()
+	cut_overlays()
+
+/** Alerts pAI cards that someone has submitted candidacy */
+/obj/item/pai_card/proc/alert_update()
+	if(!COOLDOWN_FINISHED(src, alert_cooldown))
+		return
+	COOLDOWN_START(src, alert_cooldown, 5 SECONDS)
+	add_alert()
+	addtimer(CALLBACK(src, .proc/remove_alert), 5 SECONDS)
+	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+	loc.visible_message(span_notice("[src] flashes a message across its screen: New personalities available for download!"), blind_message = span_notice("[src] vibrates with an alert."))
+
+/**
+ * Downloads a candidate from the list and removes them from SSpai.candidates
+ *
+ * @param {string} ckey The ckey of the candidate to download
+ * @returns {boolean} - TRUE if the candidate was downloaded, FALSE if not
+ */
+/obj/item/pai_card/proc/download_candidate(mob/user, ckey)
+	if(pai)
+		return FALSE
+	var/datum/pai_candidate/candidate = SSpai.candidates[ckey]
+	if(!candidate?.check_ready())
+		to_chat(user, span_warning("Download interrupted: Candidate unavailable. Please try again."))
+		return FALSE
+	var/mob/living/silicon/pai/new_pai = new(src)
+	new_pai.name = candidate.name || pick(GLOB.ninja_names)
+	new_pai.real_name = new_pai.name
+	new_pai.key = candidate.ckey
+	set_personality(new_pai)
+	SSpai.candidates.Remove(ckey)
+	return TRUE
+
+/**
+ * Pings ghosts to announce that someone is requesting a pAI
+ *
+ * @param {mob} user - The user who is requesting a pAI
+ * @returns {boolean} - TRUE if the pAI was requested, FALSE if not
+ */
+/obj/item/pai_card/proc/find_pai(mob/user)
+	if(pai)
+		return FALSE
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_SILICONS))
+		to_chat(user, span_warning("Due to growing incidents of SELF corrupted independent artificial intelligences, freeform personality devices have been temporarily	banned in this sector."))
+		return FALSE
+	if(request_spam)
+		to_chat(user, span_warning("Request sent too recently."))
+		return FALSE
+	request_spam = TRUE
+	playsound(src, 'sound/machines/ping.ogg', 20, TRUE)
+	to_chat(user, span_notice("You have requested pAI assistance."))
+	var/mutable_appearance/alert_overlay = mutable_appearance('icons/obj/aicards.dmi', "pai")
+	notify_ghosts("[user] is requesting a pAI personality! Use the pAI button to submit yourself as one.", source = user, alert_overlay = alert_overlay, action = NOTIFY_ORBIT, flashwindow = FALSE, header = "pAI Request!", ignore_key = POLL_IGNORE_PAI)
+	addtimer(CALLBACK(src, .proc/request_again), SPAM_TIME,	TIMER_UNIQUE | TIMER_STOPPABLE | TIMER_CLIENT_TIME | TIMER_DELETE_ME)
```

Comment:
consider using VARSET_CALLBACK instead, since it removes the need of defining your own function for setting a variable

Reply:
nice suggestion - didnt know about this